src/plugins/discover/public/components/common/error_callout.tsx

```diff
@@ -16,48 +23,73 @@ import { useDiscoverServices } from '../../hooks/use_discover_services';
 interface Props {
   title: string;
   error: Error;
+  isEsqlMode?: boolean;
 }
 
-export const ErrorCallout = ({ title, error }: Props) => {
-  const { core } = useDiscoverServices();
+export const ErrorCallout = ({ title, error, isEsqlMode }: Props) => {
+  const { core, docLinks } = useDiscoverServices();
   const { euiTheme } = useEuiTheme();
 
   const searchErrorDisplay = renderSearchError(error);
 
   return (
     <EuiEmptyPrompt
-      iconType="error"
-      color="danger"
+      icon={<EuiIcon size="l" type="error" color="danger" />}
+      color="plain"
+      paddingSize="m"
+      css={css`
+        margin: ${euiTheme.size.xl} auto;
+      `}
       title={
         <h2 data-test-subj="discoverErrorCalloutTitle">{searchErrorDisplay?.title ?? title}</h2>
       }
+      titleSize="xs"
+      hasBorder
       actions={searchErrorDisplay?.actions ?? []}
       body={
-        <div
-          css={css`
-            text-align: left;
-          `}
-        >
+        <>
           {searchErrorDisplay?.body ?? (
             <>
-              <p
+              <EuiCodeBlock
+                paddingSize="s"
+                language="json"
+                isCopyable
                 css={css`
-                  white-space: break-spaces;
-                  word-wrap: break-word;
-                  font-family: ${euiTheme.font.familyCode};
+                  text-align: left;
                 `}
                 data-test-subj="discoverErrorCalloutMessage"
               >
                 {error.message}
-              </p>
-              <EuiButton onClick={() => core.notifications.showErrorDialog({ title, error })}>
-                {i18n.translate('discover.errorCalloutShowErrorMessage', {
-                  defaultMessage: 'View details',
-                })}
-              </EuiButton>
+              </EuiCodeBlock>
+              {!isEsqlMode && (
+                <EuiButton
+                  onClick={() => core.notifications.showErrorDialog({ title, error })}
+                  data-test-subj="discoverErrorCalloutShowDetailsButton"
+                >
+                  {i18n.translate('discover.errorCalloutShowErrorMessage', {
+                    defaultMessage: 'View details',
+                  })}
+                </EuiButton>
+              )}
             </>
           )}
-        </div>
+        </>
+      }
+      footer={
+        <>
+          {isEsqlMode && (
+            <EuiButtonEmpty
+              iconType="documentation"
+              href={docLinks.links.query.queryESQL}
+              data-test-subj="discoverErrorCalloutESQLReferenceButton"
+              target="_blank"
+            >
+              {i18n.translate('discover.errorCalloutESQLReferenceButtonLabel', {
+                defaultMessage: 'Open ES|QL reference',
+              })}
+            </EuiButtonEmpty>
+          )}
+        </>
       }
```

Comment:
To hide the footer for data view mode

<img width="954" alt="Screenshot 2024-09-02 at 11 05 06" src="https://github.com/user-attachments/assets/289a91ea-6673-4aea-883f-83300e9be911">


```suggestion
      footer={isEsqlMode ? (
            <EuiButtonEmpty
              iconType="documentation"
              href={docLinks.links.query.queryESQL}
              data-test-subj="discoverErrorCalloutESQLReferenceButton"
              target="_blank"
            >
              {i18n.translate('discover.errorCalloutESQLReferenceButtonLabel', {
                defaultMessage: 'Open ES|QL reference',
              })}
            </EuiButtonEmpty>
          ) : undefined
      }
```
_not formatted_

Reply:
Done here https://github.com/elastic/kibana/pull/191320/commits/5c4fdf0f0df77163c422d6c21aea05d5ce0a3154